Sources/swift-build-prebuilts/main.swift

```diff
@@ -0,0 +1,437 @@
+    if dockeronly {
+        return platform.os == .linux
+    }
+    #if os(macOS)
+    if platform.os == .macos {
+        return true
+    }
+    #elseif os(Windows)
+    if platform.os == .windows {
+        return true
+    }
+    #endif
+    return docker && platform.os == .linux
+}
+
+func shell(_ command: String) throws {
+    let process = Process()
+#if os(Windows)
+    process.executableURL = URL(fileURLWithPath: "C:\\Windows\\System32\\cmd.exe")
+    process.arguments = ["/c", command]
+#else
+    process.executableURL = URL(fileURLWithPath: "/bin/bash")
+    process.arguments = ["-c", command]
+#endif
+    print("Running:", command)
+    try process.run()
+    process.waitUntilExit()
+    if process.terminationStatus != 0 {
+        throw StringError("failed: \(command)")
+    }
+}
+
+func downloadManifest(version: PrebuiltRepos.Version) async throws -> Workspace.PrebuiltsManifest? {
+    let manifestFile = swiftVersion + "-manifest.json"
+    let destination = await stageDir / manifestFile
+    if destination.exists {
+        do {
+            return try JSONDecoder().decode(
+                Workspace.PrebuiltsManifest.self,
+                from: try destination.read()
+            )
+        } catch {
+            // redownload it
+            try destination.remove()
+        }
+    }
+    let manifestURL = manifestHost.appending(components: version.tag, manifestFile)
+    print("Downloading:", manifestURL.absoluteString)
+    let httpClient = HTTPClient()
+    var headers = HTTPClientHeaders()
+    headers.add(name: "Accept", value: "application/json")
+    var request = HTTPClient.Request(kind: .generic(.get), url: manifestURL)
+    request.options.validResponseCodes = [200]
+
+    let response = try? await httpClient.execute(request) { _, _ in }
+    if let body = response?.body {
+        return try JSONDecoder().decode(
+            Workspace.PrebuiltsManifest.self,
+            from: body
+        )
+    }
+
+    return nil
+}
+
+// Path utility
+struct Path: CustomStringConvertible {
+    private let drive: Substring?
+    private let components: [Substring]
+
+    private static var separator: String {
+#if os(Windows)
+            "\\"
+#else
+            "/"
+#endif
+    }
+
+    private init(drive: Substring?, components: [Substring]) {
+        self.drive = drive
+        self.components = components
+    }
+
+    init(_ value: String) {
+#if os(Windows)
+        let winValue = value.replacingOccurrences(of: "/", with: "\\")
+        let driveComps = winValue.split(separator: ":")
+        if driveComps.count > 1 {
+            self.drive = driveComps[0]
+            self.components = driveComps[1].split(separator: "\\")
+            return
+        }
+#endif
+        self.drive = nil
+        self.components = value.split(separator: Self.separator)
+    }
+
+    var path: String {
+#if os(Windows)
+        if let drive {
+            return drive + ":\\" + components.joined(separator: "\\")
```

Comment:
Nit: use `seperator` instead of the two backslashes

Reply:
I've adopted AbsolutePath instead.